pkgs/applications/misc/johnny-reborn/with-data.nix

```diff
@@ -0,0 +1,55 @@
+{ lib, stdenv, fetchFromGitHub, fetchzip, makeWrapper, johnny-reborn_engine }:
+stdenv.mkDerivation {
+  pname = "johnny-reborn";
+  version = johnny-reborn_engine.version;
+
+  srcs =
+    let
+      sounds = fetchFromGitHub {
+        owner = "nivs1978";
+        repo = "Johnny-Castaway-Open-Source";
+        rev = "be6afefd43a3334acc66fc9d777c162c8bfb9558";
+        hash = "sha256-rtZVCn4KbEBVwaSQ4HZhMoDEI5Q9IPj9SZywgAx0MPY=";
+      };
+
+      resources = fetchzip {
+        name = "scrantic-source";
+        url = "https://archive.org/download/johnny-castaway-screensaver/scrantic-run.zip";
+        hash = "sha256-Q9chCYReOQEmkTyIkYo+D+OXYUqxPNOOEEmiFh8yaw4=";
+        stripRoot = false;
+      };
+    in
+    [
+      sounds
+      resources
+    ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  sourceRoot = "source";
+
+  dontBuild = true;
+
+  installPhase = ''
```

Comment:
A hint: I believe this style of derivation can be made similar to Commander X16 package, especially `run.nix`.

Reply:
It's the same thing with extra steps since their `runScript` becomes a derivation, their resources are in another derivation, and everything is symlinked to a fourth derivation. Meanwhile:
1. My solutions keep everything under 2 derivations;
2. There are a total of ZERO symlinks, everything is in its right place;
3. If I had "second" and "third" derivations like that, nobody else would be re-using it, the biggest advantage of saving disk-storage would be unprofitable.